apps/angular/ngfor-enhancement/src/app/app.component.ts

```diff
@@ -7,15 +6,13 @@ interface Person {
 
 @Component({
   standalone: true,
-  imports: [NgFor, NgIf],
   selector: 'app-root',
   template: `
-    <ng-container *ngIf="persons.length > 0; else emptyList">
-      <div *ngFor="let person of persons">
-        {{ person.name }}
-      </div>
-    </ng-container>
-    <ng-template #emptyList>The list is empty !!</ng-template>
+    @for (person of persons; track person.name) {
+      <div>{{ person.name }}</div>
+    } @empty {
+      The list is empty !!
+    }
```

Comment:
ahah, This challenge is becoming obsolete with the new control flow. I should refactor it to enhance another directive. Enhancing directive is still a good knowlegde to have. Structural directive are not going anywhere soon. 

But for this challenge, this solution works